code/game/gamemodes/gang/dominator.dm

```diff
@@ -0,0 +1,225 @@
+				takeover_in_progress = 1
+				break
+		if(!takeover_in_progress)
+			shuttle_master.emergencyNoEscape = 0
+			if(shuttle_master.emergency.mode == SHUTTLE_STRANDED)
+				shuttle_master.emergency.mode = SHUTTLE_DOCKED
+				shuttle_master.emergency.timer = world.time
+				command_announcement.Announce("Hostile enviroment resolved. You have 3 minutes to board the Emergency Shuttle.", null, 'sound/AI/shuttledock.ogg', "Priority")
+			else
+				command_announcement.Announce("All hostile activity within station systems have ceased.","Network Alert")
+
+			if(get_security_level() == "delta")
+				set_security_level("red")
+
+		gang.message_gangtools("Hostile takeover cancelled: Dominator is no longer operational.[gang.dom_attempts ? " You have [gang.dom_attempts] attempt remaining." : " The station network will have likely blocked any more attempts by us."]",1,1)
+
+	luminosity = 0
+	icon_state = "dominator-broken"
+	overlays.Cut()
+	operating = -1
+	processing_objects.Remove(src)
+
+/obj/machinery/dominator/Destroy()
+	if(operating != -1)
+		set_broken()
+	return ..()
+
+/obj/machinery/dominator/emp_act(severity)
+	healthcheck(100)
+	..()
+
+/obj/machinery/dominator/ex_act(severity, target)
+	if(target == src)
+		qdel(src)
+		return
+	switch(severity)
+		if(1)
+			qdel(src)
+		if(2)
+			healthcheck(120)
+		if(3)
+			healthcheck(30)
+	return
+
+/obj/machinery/dominator/bullet_act(obj/item/projectile/Proj)
+	if(Proj.damage)
+		if((Proj.damage_type == BRUTE || Proj.damage_type == BURN))
+			var/damage = Proj.damage
+			if(Proj.forcedodge)
+				damage *= 0.5
+			playsound(src, 'sound/effects/bang.ogg', 50, 1)
+			visible_message("<span class='danger'>[src] was hit by [Proj].</span>")
+			healthcheck(damage)
+	..()
+
+/obj/machinery/dominator/blob_act()
+	healthcheck(110)
+
+/obj/machinery/dominator/attackby(obj/I, mob/user, params)
+
+	return
+
+/obj/machinery/dominator/attack_hand(mob/user)
+	if(operating)
+		examine(user)
+		return
+
+	var/datum/gang/tempgang
+
+	if(user.mind in ticker.mode.get_all_gangsters())
+		tempgang = user.mind.gang_datum
+	else
+		examine(user)
+		return
+
+	if(isnum(tempgang.dom_timer))
+		user << "<span class='warning'>Error: Hostile Takeover is already in progress.</span>"
+		return
+
+	if(!tempgang.dom_attempts)
+		user << "<span class='warning'>Error: Unable to breach station network. Firewall has logged our signature and is blocking all further attempts.</span>"
+		return
+
+	var/time = round(get_domination_time(tempgang)/60,0.1)
+	if(alert(user,"With [round((tempgang.territory.len/start_state.num_territories)*100, 1)]% station control, a takeover will require [time] minutes.\nYour gang will be unable to gain influence while it is active.\nThe entire station will likely be alerted to it once it starts.\nYou have [tempgang.dom_attempts] attempt(s) remaining. Are you ready?","Confirm","Ready","Later") == "Ready")
+		if (isnum(tempgang.dom_timer) || !tempgang.dom_attempts || !in_range(src, user) || !istype(src.loc, /turf))
```

Comment:
This should really have checks for user conciousness / restraints, and adjacency.